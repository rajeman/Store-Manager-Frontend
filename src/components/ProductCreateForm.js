import React from 'react';
import { connect } from 'react-redux';
import { createProduct, setProductError } from '../actions/products';

class ProductCreateForm extends React.Component {
  componentDidMount() {
    this.props.dispatch(setProductError(''));
  }
  render() {
    const { productCreate, productCreateError } = this.props.products;
    const onFormSubmit = (e) => {
      e.preventDefault();
      const name = e.target.elements.pname.value.trim();
      const quantity = e.target.elements.pquantity.value;
      const price = e.target.elements.pprice.value;
      const minInvent = e.target.elements.pinventory.value;
      this.props.dispatch(createProduct(name, quantity, price, minInvent));
    }
    return (
      <div className="cont">
        <section>
          <div id="user-form">
            <div className="container">
              <form onSubmit={onFormSubmit} >
                <h3>New Product</h3>
                <div>
                  <label>Name:</label>
                  <br />
                  <input type="text" placeholder="Enter Product Name" name="pname" required></input>
                </div>
                <div>
                  <label>Quantity:</label>
                  <br />
                  <input type="number" placeholder="Enter Product Quantity" name="pquantity" required></input>
                </div>
                <div>
                  <label>Price:</label>
                  <br />
                  <input type="text" placeholder="Enter Product Price" name="pprice" required></input>
                </div>
                <div>
                  <label>Minimum Inventory:</label>
                  <br />
                  <input type="text" placeholder="Enter Minimum Inventory" name="pinventory" required></input>
                </div>
                <button type="submit">Create</button>
                <br />
                <div className="not-registered">
                  {productCreate === 'STATE_CREATE_FAILED' && <span className="pop-up">{productCreateError}</span>}
                  {productCreate === 'STATE_CREATING' && <span className="lds-hourglass"></span>}
                </div>
              </form>
            </div>
          </div>
        </section>
      </div>
    );
  }
}

const mapStateToProps = (state) => state;

export default connect(mapStateToProps)(ProductCreateForm);

