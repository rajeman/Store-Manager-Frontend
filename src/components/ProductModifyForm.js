import React from 'react';
import { connect } from 'react-redux';
import { modifyProduct, setModifyError, fetchProduct } from '../actions/products';

class ProductModifyForm extends React.Component {
  componentDidMount() {
    this.props.dispatch(setModifyError(''));
    this.props.dispatch(fetchProduct(this.props.id));
  }

  render() {
    const { product, productModify, productModifyError } = this.props.products;
    const onFormSubmit = (e) => {
      e.preventDefault();
      const name = e.target.elements.pname.value.trim();
      const quantity = e.target.elements.pquantity.value;
      const price = e.target.elements.pprice.value;
      const minInvent = e.target.elements.pinventory.value;
      this.props.dispatch(modifyProduct(name, quantity, price, minInvent, this.props.id));
    }
    if (product[0]) {
      return (
        <div className="cont">
          <section>
            <div id="user-form">
              <div className="container">
                <form onSubmit={onFormSubmit}>
                  <h3>Modify Product</h3>
                  <div>
                    <label>Name:</label>
                    <br />
                    <input type="text" placeholder="Enter Product Name" name="pname" pattern=".{3,}" required title="Name must be at least 3 characters" defaultValue={product[0].product_name} ></input>
                  </div>
                  <div>
                    <label>Quantity:</label>
                    <br />
                    <input type="text" placeholder="Enter Product Quantity" name="pquantity" pattern="^\+?[1-9][\d]*$" required title="Quantity must be a positive integer" defaultValue={product[0].product_quantity}></input>
                  </div>
                  <div>
                    <label>Price:</label>
                    <br />
                    <input type="text" placeholder="Enter Product Price" name="pprice" pattern="^\+?[1-9][\d]*$" required title="Price must be a positive integer" defaultValue={product[0].product_price}></input>
                  </div>

                  <div>
                    <label>Minimum Inventory:</label>
                    <br />
                    <input type="text" placeholder="Enter Minimum Inventory" name="pinventory" required defaultValue={product[0].minimum_inventory}></input>
                  </div>
                  <button type="submit">Modify</button>
                  <br />
                  <div className="not-registered">
                    {productModify === 'STATE_MODIFY_FAILED' && <span className="pop-up">{productModifyError}</span>}
                    {productModify === 'STATE_MODIFYING' && <span className="lds-hourglass"></span>}
                  </div>
                </form>
              </div>
            </div>
          </section>
        </div>
      );
    } else {
      return <div></div>
    }
  }
}

const mapStateToProps = (state) => state;

export default connect(mapStateToProps)(ProductModifyForm);


