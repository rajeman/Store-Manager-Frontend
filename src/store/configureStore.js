import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import productReducer from '../reducers/products';
import authReducer from '../reducers/auth';
import saleReducer from '../reducers/records';
import navigationReducer from '../reducers/navigation';
import attendantReducer from '../reducers/attendants';
import cartReducer from '../reducers/cart';

export default () => createStore(
  combineReducers({
    products: productReducer,
    auth: authReducer,
    navigation: navigationReducer,
    sales: saleReducer,
    attendants: attendantReducer,
    cart: cartReducer,
  }), applyMiddleware(thunk),
);
