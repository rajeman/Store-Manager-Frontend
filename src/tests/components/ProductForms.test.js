import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import axios from 'axios';
import ProductModifyForm from '../../components/ProductModifyForm';
import ProductCreateForm from '../../components/ProductCreateForm';

jest.mock('axios');

describe('PRODUCT FORMS TEST SUITE', () => {
  describe('PRODUCT MODIFY FORM TEST SUITE', () => {
    it('should render the ProductModify form', () => {
      const response = {
        data:
        {
          message: ''
        }
      };
      axios.get.mockResolvedValue(response);
      const initialState = {
        products: {
          product: []
        },
      };
      const mockStore = configureStore([thunk]);
      const store = mockStore(initialState);
      const component = mount(
        <Provider store={store}>
          <ProductModifyForm />
        </Provider>
      );
      expect(component).toMatchSnapshot();
    });

    it('should make request to modify product when form is submitted', () => {
      const response = {
        data:
        {
          message: 'successfully modified product'
        }
      };
      axios.put.mockImplementation(() => Promise.resolve(response));
      const initialState = {
        products: {
          product: [{
            id: 34,
            name: "Arduino Uno"
          }]
        },
      };
      const mockStore = configureStore([thunk]);
      const store = mockStore(initialState);
      const component = mount(
        <Provider store={store}>
          <ProductModifyForm />
        </Provider>
      );
      component.find('form').simulate('submit', {});
      const dispatchedActions = store.getActions();
      expect(dispatchedActions[1].type).toEqual('PRODUCT_MODIFY');
    });
  });

  describe('PRODUCT CREATE FORM TEST SUITE', () => {
    it('should render the ProductCreate Form', () => {
      const initialState = {
        products: {
          productCreate: 'STATE_CREATE_FAILED',
          productCreateError: ''
        }
      };
      const mockStore = configureStore([thunk]);
      const store = mockStore(initialState);
      const component = mount(
        <Provider store={store}>
          <ProductCreateForm />
        </Provider>
      );
      expect(component).toMatchSnapshot();
    });

    it('should make request to create product when form is submitted', () => {
      const response = {
        data:
        {
          message: 'successfully modified product'
        }
      };
      axios.post.mockImplementation(() => Promise.resolve(response));
      const initialState = {
        products: {
          productCreate: 'STATE_CREATING',
          productCreateError: ''
        }
      };
      const mockStore = configureStore([thunk]);
      const store = mockStore(initialState);
      const component = mount(
        <Provider store={store}>
          <ProductCreateForm />
        </Provider>
      );
      component.find('form').simulate('submit', {});
      const dispatchedActions = store.getActions();
      expect(dispatchedActions[1].productCreate).toEqual('STATE_CREATING');
    });
  });
});


