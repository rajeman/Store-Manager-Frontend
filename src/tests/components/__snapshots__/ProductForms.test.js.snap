// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PRODUCT FORMS TEST SUITE PRODUCT MODIFY FORM TEST SUITE should render the ProductModify form 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ProductModifyForm)>
    <ProductModifyForm
      dispatch={[Function]}
      products={
        Object {
          "product": Array [],
        }
      }
    >
      <div />
    </ProductModifyForm>
  </Connect(ProductModifyForm)>
</Provider>
`;
